src/datepicker/datepicker.ts

```diff
@@ -117,7 +117,8 @@ export interface NgbDatepickerNavigateEvent {
 
     <ng-template [ngTemplateOutlet]="footerTemplate"></ng-template>
   `,
-  providers: [NGB_DATEPICKER_VALUE_ACCESSOR, NgbDatepickerService, NgbDatepickerKeyMapService]
+  providers:
+      [NGB_DATEPICKER_VALUE_ACCESSOR, NgbDatepickerService, NgbDatepickerKeyMapService, NgbDatepickerKeyMapperService]
```

Comment:
Hmm, am I mistaken, but how are you going to override `NgbDatepickerKeyMapperService`, if you're providing it locally ?